optax/_src/alias_test.py

```diff
@@ -22,80 +22,85 @@
 
-  @parameterized.parameters(
-      ('sgd', lambda: alias.sgd(1e-2, 0.0)),
-      ('adam', lambda: alias.adam(1e-1)),
-      ('adamw', lambda: alias.adamw(1e-1)),
-      ('lamb', lambda: alias.adamw(1e-1)),
-      ('rmsprop', lambda: alias.rmsprop(1e-1)),
-      ('rmsprop_momentum', lambda: alias.rmsprop(5e-2, momentum=0.9)),
-      ('fromage', lambda: alias.fromage(1e-2)),
-      ('adabelief', lambda: alias.adabelief(1e-1)),
-      ('radam', lambda: alias.radam(1e-1)),
-      ('sm3', lambda: alias.sm3(1.0)),
-      ('yogi', lambda: alias.yogi(1.0)),
-      ('dpsgd', lambda: alias.dpsgd(1e-2, 10.0, 0.001, 0))
-  )
-  def test_parabel(self, opt_name, opt):
-    opt = opt()
+  if jnp.iscomplexobj(dtype):
+    final_params *= 1 + 1j
 
-    initial_params = jnp.array([-1.0, 10.0, 1.0])
-    final_params = jnp.array([1.0, -1.0, 1.0])
+  @jax.grad
+  def get_updates(params):
+    return jnp.sum(numerics.abs2(params - final_params))
 
-    @jax.grad
-    def get_updates(params):
-      return jnp.sum((params - final_params)**2)
+  return initial_params, final_params, get_updates
 
-    @jax.jit
-    def step(params, state):
-      updates = get_updates(params)
-      if opt_name == 'dpsgd': updates = updates[None]
-      updates, state = opt.update(updates, state, params)
-      params = update.apply_updates(params, updates)
-      return params, state
 
-    params = initial_params
-    state = opt.init(params)
-    for _ in range(1000):
-      params, state = step(params, state)
+def _setup_rosenbrock(dtype):
+  """Rosenbrock function as an optimization target."""
+  a = 1.0
+  b = 100.0
 
-    chex.assert_tree_all_close(params, final_params, rtol=1e-2, atol=1e-2)
-
-  @parameterized.parameters(
-      ('sgd', lambda: alias.sgd(2e-3, 0.2)),
-      ('adam', lambda: alias.adam(1e-1)),
-      ('adamw', lambda: alias.adamw(1e-1)),
-      ('lamb', lambda: alias.adamw(1e-1)),
-      ('rmsprop', lambda: alias.rmsprop(5e-3)),
-      ('rmsprop_momentum', lambda: alias.rmsprop(5e-3, momentum=0.9)),
-      ('fromage', lambda: alias.fromage(5e-3)),
-      ('adabelief', lambda: alias.adabelief(1e-1)),
-      ('radam', lambda: alias.radam(1e-3)),
-      ('sm3', lambda: alias.sm3(1.0)),
-      ('yogi', lambda: alias.yogi(1.0)),
-      ('dpsgd', lambda: alias.dpsgd(2e-3, 10., 0.001, 0, 0.2))
-  )
-  def test_rosenbrock(self, opt_name, opt):
-    opt = opt()
+  if jnp.iscomplexobj(dtype):
+    a *= 1 + 1j
+
+  initial_params = jnp.array([0.0, 0.0], dtype=dtype)
+  final_params = jnp.array([a, a**2], dtype=dtype)
+
+  @jax.grad
+  def get_updates(params):
+    return (numerics.abs2(a - params[0]) +
+        b * numerics.abs2(params[1] - params[0]**2))
+
+  return initial_params, final_params, get_updates
 
-    a = 1.0
-    b = 100.0
-    initial_params = jnp.array([0.0, 0.0])
-    final_params = jnp.array([a, a**2])
 
-    @jax.grad
-    def get_updates(params):
-      return (a - params[0])**2 + b * (params[1] - params[0]**2)**2
+class AliasTest(chex.TestCase):
```

Comment:
(nit, feel free to ignore this comment) 
This test derives from chex.TestCase but doesn't use variants which can improve the coverage: https://github.com/deepmind/chex#test-variants-variantspy